runtime/jvmti/jvmtiStartup.c

```diff
@@ -1062,10 +1061,56 @@ createXrunLibraries(J9JavaVM * vm)
 	return JNI_OK;
 }
 
+/**
+ * Test if an agent library was already loaded.
+ *
+ * @param vm Java VM
+ * @param library library name
+ * @param decorate a boolean indicating if the prefix/suffix to the library name is to be added
+ *
+ * @return TRUE if the agent library was loaded successfully, FALSE otherwise
+ */
 static BOOLEAN
-isAgentLibraryLoaded(J9JavaVM *vm, const char *library)
+isAgentLibraryLoaded(J9JavaVM *vm, const char *library, BOOLEAN decorate)
 {
-	return NULL != findAgentLibrary(vm, library, strlen(library));
+	BOOLEAN result = FALSE;
+
+	if (NULL == findAgentLibrary(vm, library, strlen(library))) {
+		/* J9PORT_SLOPEN_NO_LOAD is only supported on Linux/OSX/Win32 platforms. */
+#if defined(LINUX) || defined(OSX) || defined(WIN32)
+		int i = 0;
+		const char *systemAgentName = NULL;
+		UDATA handle = 0;
+		/* One of RTLD_LAZY and RTLD_NOW must be included in the flag. */
+		UDATA openFlags = J9PORT_SLOPEN_NO_LOAD | OMRPORT_SLOPEN_LAZY;
+		char *agentPath = (char*)library;
+		PORT_ACCESS_FROM_JAVAVM(vm);
+
+		if (decorate) {
+			openFlags |= J9PORT_SLOPEN_DECORATE;
+		}
+		while (NULL != (systemAgentName = systemAgentNames[i++])) {
+			if (0 == strcmp(library, systemAgentName)) {
+				agentPath = prependSystemAgentPath(vm, library);
+				break;
+			}
+		}
```

Comment:
Assignments, increments and tests should all be separate:
```c
		for (i = 0;; ++i) {
			const char *systemAgentName = systemAgentNames[i];
			if (NULL == systemAgentName) {
				break;
			}
			if (0 == strcmp(library, systemAgentName)) {
				agentPath = prependSystemAgentPath(vm, library);
				break;
			}
		}
```

Reply:
Updated